core/sequencer/server.go

```diff
@@ -527,8 +527,9 @@ func (s *Server) PublishRevisions(ctx context.Context,
 		end = batch
 	}
 	logRootTrail.Set(float64(latestMapRoot.Revision - logRoot.TreeSize))
-	for rev := logRoot.TreeSize - 1; rev <= end; rev++ {
-		rawMapRoot, mapRoot, err := mapClient.GetAndVerifyMapRootByRevision(ctx, int64(rev))
+	maxLeafIndex := logRoot.TreeSize - 1
+	for revToWrite := maxLeafIndex + 1; revToWrite <= end; revToWrite++ {
```

Comment:
```suggestion
	for revToWrite := logRoot.TreeSize; revToWrite <= end; revToWrite++ {
```

Reply:
I know this has fewer lines, but since the math confused me for a hot second, I thought I'd split it out as a favor to future readers.

Reply:
The new math does not seem less confusing to me. In fact, it triggers some questions: "hm, this line computes `logRoot.TreeSize - 1`. Does this mean it expects `TreeSize` > 0? Aaaah, there is +1 in the next line, so it's fine". I think what happened is that you got to the bottom of the bug, and got a false sense that changing this code will improve readability. IMO, what would really help to clarify things is a comment (akin to this PR description), and the actual code can be as short as possible.